pipeline/runtime/runtestsV2.go

```diff
@@ -0,0 +1,330 @@
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", dir))
+		return "", err
+	}
+
+	iniFileDir := fmt.Sprintf("%s/new", tmpDir)
+	err = fs.MkdirAll(iniFileDir, os.ModePerm)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", iniFileDir))
+		return "", err
+	}
+	iniFile := fmt.Sprintf("%s/config.ini", iniFileDir)
+	data := fmt.Sprintf(`outDir: %s
+	logLevel: 0 
+	logConsole: false
+	writeTo: JSON
+	packageInference: true`, dir)
+
+	log.Infof("Attempting to write to %s with config:\n%s", iniFile, data)
+	f, err := fs.Create(iniFile)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create config file %s", iniFile))
+		return "", err
+	}
+
+	_, err = f.WriteString(data)
+	defer f.Close()
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not write %s to config file %s", data, iniFile))
+		return "", err
+	}
+
+	return iniFile, nil //path of config.ini file
+}
+
+func getPreCmd(tmpFilePath string, fs filesystem.FileSystem, log *logrus.Logger) (string, error) {
+	var preCmd string
+	iniFilePath, err := createJavaConfigFile(tmpFilePath, fs, log)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create java agent config file in path %s", iniFilePath))
+		return "", err
+	}
+
+	err = writetoBazelrcFile(iniFilePath, log, fs, tmpFilePath)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("failed to write in .bazelrc file"))
+		return "", err
+	}
+	javaAgentPath := fmt.Sprintf("%s%s%s", tmpFilePath, javaNewAgentPath, javaNewAgentJar)
+	agentArg := fmt.Sprintf(javaNewAgentArg, javaAgentPath, iniFilePath)
+	preCmd = fmt.Sprintf("export JAVA_TOOL_OPTIONS=%s", agentArg)
+	return preCmd, nil
+}
+
+func downloadJavaAgent(ctx context.Context, path string, fs filesystem.FileSystem, log *logrus.Logger) error {
+
+	javaAgentPath := fmt.Sprintf("%s%s", javaNewAgentPath, javaNewAgentJar)
+	dir := filepath.Join(path, javaAgentPath)
+	err := utils.DownloadFile(ctx, dir, javaNewAgentUrl, fs)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not download java agent"))
+		return err
+	}
+	return nil
+}
+
+func createFilterFile(ctx context.Context, fs filesystem.FileSystem, stepID, workspace string, log *logrus.Logger,
+	tiConfig *tiCfg.Cfg, path string) error {
+
+	isManualExecution := utils.IsManualExecution(tiConfig)
+	resp, isFilterFilePresent := getTestsSelection(ctx, fs, stepID, workspace, log, isManualExecution, tiConfig)
+	dir := filepath.Join(path, filterDir)
+	err := fs.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", dir))
+		return err
+	}
+	err = populateItemInFilterFile(resp, dir, fs, isFilterFilePresent)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func writetoBazelrcFile(iniFilePath string, log *logrus.Logger, fs filesystem.FileSystem, tmpFilePath string) error {
```

Comment:
Should we look for bazelrc file in the test command? Users can give another file as an input. You can refer to PR checks pipeline.

Reply:
In order to support for bazel, env variable had to be exported in .bazelrc file else overridden of env var takes place which is being used to inject java agent. We can have this support in .bazelrc in home directory level. @smjt-h had tested this case.